### ISSUE TITLE
acf-block-preview class missing when block is initially loaded in edit mode.

### ISSUE THREAD

**Issue by will-yellowpeach** [AdvancedCustomFields/acf#620]
Since the latest update, we're having an issue with the ACF block preview mode. By default, ACF adds .acf-block-preview to each block when in preview mode, and we use that class to ensure our CSS doesn't effect the edit mode fields. However the class doesn't always seem to be there. 

Steps to reproduce:

1. Add an ACF block to a post
2. Switch the block to Edit mode and save the post.
3. Refresh the page and then go back to the to the block and toggle to Preview mode - you'll notice the .acf-block-preview class is not there in this instance.

It appears that if blocks are initialised in Preview mode then the class is there, but if they are initialised in Edit mode and then toggled, the class does not show.

**Comment by lgladdy** [comment]
Thanks for the report @will-yellowpeach. Just to confirm are you saying this was changed in ACF 5.12? Or did you update from an earlier version of ACF than 5.11?

**Comment by will-yellowpeach** [comment]
Yes this has only started happening in 5.12 - we've noticed it on all our Gutenberg custom builds.

**Comment by lgladdy** [comment]
@will-yellowpeach Thanks for confirming. I recall us making a change for this in an earlier ACF build - but we also disabled edit mode preloading in ACF 5.12 due to a last-minute bug introduced by WordPress 5.9.1, so it's possible this bug was introduced by that. We'll take a look! Cheers.

**Comment by lgladdy** [comment]
Hey @will-yellowpeach. I've taken a quick look into this and can't reproduce a change into ACF 5.12

There was a bug in ACF 5.10's release which was fixed in a minor revision (5.10.1 or 5.10.2), as .acf-block-preview is not supposed to be present for edit mode forms.

Could you provide some more information if I'm misunderstanding the issue please?

**Comment by will-yellowpeach** [comment]
Hi @lgladdy . Here is a screen recording which hopefully helps explain the issue: https://drive.google.com/file/d/1GQpJXYI4VnAd89szGWF3-ZOD0wWDIXHu/view?usp=sharing.

You'll see the block is initially loaded in Preview mode and the .acf-block-preview class is there and the example heading is displaying correctly (it has the custom font).

Then I switched to Edit mode, saved the page and refreshed. You can then see when I switched back to Edit mode the heading was using the default WP font because the .acf-block-preview class was not there. Weirdly, when I toggle to Edit and back to Preview for a second time, the class does seem to then get loaded after a second or so which you'll see in the video.

**Comment by lgladdy** [comment]
Thanks for the video @will-yellowpeach, so I think we're mixing up Edit and preview mode here...

The issue is that, after loading a page with a block in edit mode, when you switch to preview the block, the `acf-block-preview` class isn't there initially (until an AJAX load happens and it kicks in again) right?

You're not expecting the edit form (with the input fields) to have the `acf-block-preview` class right?

**Comment by will-yellowpeach** [comment]
Yes that is correct. Sorry, may have confused things slightly! The reason I referred to edit mode in the issue title was because the issue only occurs when the block is initialised in Edit mode and not in Preview mode.

But no, we're not expecting the .acf-block-preview class in edit - all our CSS is wrapped in that class specifically so we don't target the ACF fields that are in edit mode

**Comment by lgladdy** [comment]
@will-yellowpeach No worries! Agreed whatever is happening in your video is definitely a bug! I'll try and reproduce it here and get back to you for more info if I can't reproduce it, cheers!

**Comment by lgladdy** [comment]
@will-yellowpeach Hey, I'd hope if we can nail this down we'll get this into our next minor release as it fixes a bug introduced in a major, so it would be a couple of weeks max.

That said; I can't reproduce this locally. For me, when I click the preview button, an AJAX load happens and the AJAX result correctly contains the acf-block-preview class.

Can you watch the network tab when you switch back to the preview class and encounter the bug and see if an AJAX load happens for you and the result of this content? If not, that would suggest something in block preloading is going wrong. In that case: Do you have more than one of the block on the page?

Alternatively, if you have this site up on staging somewhere you could give me access to - I'll be happy to debug on your staging site directly. You can drop the details over to `support@advancedcustomfields.com` and we'll pick it up there.

**Comment by will-yellowpeach** [comment]
Hi @lgladdy, no AJAX request being made when I initially toggle to Preview mode, no. However when I toggle back to edit and then to Preview again for a second time, the AJAX request is then made. Video here showing that: https://drive.google.com/file/d/1E_UAKmxrTXSFsNR3yoeRDLZ1yGAeWHSc/view?usp=sharing

This is the only block we have on the page - tried it with other blocks we have built too, and on a new page to make sure nothing is corrupt but the same issue seems to persist.

We don't have a staging site set up for this one just yet, but if you're struggling to replicate happy to jump on a quick Zoom call or we can look at setting up a staging site if really needed.

**Comment by lgladdy** [comment]
@will-yellowpeach How about sending us a gist url for the full DOM content of the initial page load when the block has been saved in edit mode? There will be an `preloadedBlocks` variable inside the `acf.data` JSON in the DOM that is presumably containing the preview mode view without the right class.

If you just send us the JSON for `acf.data` that would be even better :)

**Comment by lgladdy** [comment]
@will-yellowpeach Actually scratch that. I found the bug! We'll aim to get this fixed in ACF 5.12.1 in the next couple of weeks. Sorry about this. For now, you can solve the issue by disabling block preloading in your theme or plugin, 

```
add_action( 'acf/init', 'disable_block_preloading' );
function disable_block_preloading() {
	acf_update_setting( 'preload_blocks', false );
}
```

**Comment by will-yellowpeach** [comment]
@lgladdy great thanks for the speedy support
